cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/InfoAction.hs

```diff
@@ -0,0 +1,413 @@
+import qualified Hedgehog.Extras.Stock.IO.Network.Sprocket as IO
+import           Prelude
+import           System.FilePath ((</>))
+import           Testnet.Components.SPO
+import qualified Testnet.Process.Cli as P
+import qualified Testnet.Process.Run as H
+import qualified Testnet.Property.Utils as H
+import           Testnet.Runtime
+
+
+hprop_ledger_events_info_action :: Property
+hprop_ledger_events_info_action = H.integrationRetryWorkspace 2 "info-hash" $ \tempAbsBasePath' -> do
+
+  -- Start a local test net
+  conf@Conf { tempAbsPath } <- H.noteShowM $ mkConf tempAbsBasePath'
+  let tempAbsPath' = unTmpAbsPath tempAbsPath
+      tempBaseAbsPath = makeTmpBaseAbsPath tempAbsPath
+
+  work <- H.createDirectoryIfMissing $ tempAbsPath' </> "work"
+
+  let sbe = ShelleyBasedEraConway
+      era = toCardanoEra sbe
+      cEra = AnyCardanoEra era
+      fastTestnetOptions = cardanoDefaultTestnetOptions
+        { cardanoEpochLength = 100
+        , cardanoSlotLength = 0.1
+        , cardanoNodeEra = cEra
+        }
+
+  testnetRuntime@TestnetRuntime
+    { testnetMagic
+    , poolNodes
+    , wallets
+    }
+    <- cardanoTestnet fastTestnetOptions conf
+
+  poolNode1 <- H.headM poolNodes
+
+  poolSprocket1 <- H.noteShow $ nodeSprocket $ poolRuntime poolNode1
+
+  execConfig <- H.mkExecConfig tempBaseAbsPath poolSprocket1
+
+  let socketName' = IO.sprocketName poolSprocket1
+      socketBase = IO.sprocketBase poolSprocket1 -- /tmp
+      socketPath = socketBase </> socketName'
+
+  H.note_ $ "Sprocket: " <> show poolSprocket1
+  H.note_ $ "Abs path: " <> tempAbsBasePath'
+  H.note_ $ "Socketpath: " <> socketPath
+  H.note_ $ "Foldblocks config file: " <> configurationFile testnetRuntime
+
+  gov <- H.createDirectoryIfMissing $ work </> "governance"
+  proposalAnchorFile <- H.note $ work </> gov </> "sample-proposal-anchor"
+  infoActionFp <- H.note $ work </> gov </> "info.action"
+
+  H.writeFile proposalAnchorFile "dummy anchor data"
+
+  proposalAnchorDataHash <- H.execCli' execConfig
+    [ "conway", "governance"
+    , "hash", "anchor-data", "--file-text", proposalAnchorFile
+    ]
+
+  let stakeVkeyFp = gov </> "stake.vkey"
+      stakeSKeyFp = gov </> "stake.skey"
+
+  _ <- P.cliStakeAddressKeyGen tempAbsPath'
+         $ P.KeyNames { P.verificationKeyFile = stakeVkeyFp
+                      , P.signingKeyFile = stakeSKeyFp
+                      }
+
+  let drepVkeyFp :: Int -> FilePath
+      drepVkeyFp n = gov </> "drep-keys" <>"drep" <> show n <> ".vkey"
+
+      drepSKeyFp :: Int -> FilePath
+      drepSKeyFp n = gov </> "drep-keys" <>"drep" <> show n <> ".skey"
+
+  -- Create DReps -- TODO: Refactor with shelleyKeyGen
+  H.forConcurrently_ [1..3] $ \n -> do
+   H.execCli' execConfig
+     [ "conway", "governance", "drep", "key-gen"
+     , "--verification-key-file", drepVkeyFp n
+     , "--signing-key-file", drepSKeyFp n
+     ]
+
+  -- Create Drep registration certificates
+  let drepCertFile :: Int -> FilePath
```

Comment:
This is fine for now but we can inject the dreps into the genesis after the next release of api and cli. We will probably want a single standalone test that makes sure the cli can generate the certificates and successfully submit them.